images/macos/provision/core/xcode.ps1

```diff
@@ -17,7 +17,10 @@ $env:SPACESHIP_SKIP_2FA_UPGRADE = 1
 $os = Get-OSVersion
 $xcodeVersions = Get-ToolsetValue "xcode.versions"
 $defaultXcode = Get-ToolsetValue "xcode.default"
-[Array]::Reverse($xcodeVersions)
+if ($xcodeVersions.version.Count -gt 1)
+{
+    [Array]::Reverse($xcodeVersions)
+}
```

Comment:
May be it will be easier to just change:
`$xcodeVersions = Get-ToolsetValue "xcode.versions"`
to
`[array]$xcodeVersions = Get-ToolsetValue "xcode.versions"` ?

In this case, `$xcodeVersions` will be always array and it will be easier to work with it.

Reply:
I thought the purpose of the reversing is to install Xcode starting from the lowest version, isn't it?

Reply:
In case than Count is 1:
```
PS > $xcodeVersions = Get-ToolsetValue "xcode.versions"
PS > $xcodeVersions.GetType()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     False    PSCustomObject                           System.Object
```

Reply:
yea, Alex is right. The reason of failure is that array with 1 element === 1 element and you run `Reverse` on single element instead of array with 1 element

Reply:
Thank you, folks!